### PR TITLE
Enable init/fini on CPU path

### PR DESCRIPTION
**Pull request by jbruestle**
Initial 'it does something' version of init-fini code.

### CODE REVIEW

File: pmlc/conversion/stdx_to_llvm/stdx_to_llvm.cc [hunk 1]
```diff
@@ -315,8 +315,8 @@ void populateStdXToLLVMConversionPatterns(LLVMTypeConverter &converter,
                   TanLowering,     //
                   UnpackLowering   //
                   >(converter);
-  converter.addConversion([](TupleType type) -> Optional<Type> {
-    // Tuples look like i8 pointers.  I'd like this to be void*, but MLIR
+  converter.addConversion([](stdx::ArgpackType type) -> Optional<Type> {
+    // Opaque types look like i8 pointers.  I'd like this to be void*, but MLIR
```
Comment (flaub) [inline pmlc/conversion/stdx_to_llvm/stdx_to_llvm.cc:319]: Adjust comment to say "Argpack types..."

File: pmlc/dialect/tile/transforms/BUILD [hunk 2]
```diff
@@ -22,6 +22,7 @@ plaidml_cc_library(
         "pad_ranges.cc",
         "padding.cc",
         "pass_detail.h",
+	"split_main.cc",
```
Comment (flaub) [inline pmlc/dialect/tile/transforms/BUILD:25]: whitespace

**Comment by jbruestle** [comment]
Now passes all local tests, running on CI.

**Comment by jbruestle** [comment]
This change adds support for init/fini on the CPU path only, with no additional optimizations or hoisting (so it's basically a no-op in terms of performance), but it sets the stage for adding hoisting of code from main -> init (or for alloc/dealloc ops from main -> init/fini).  The next commits will be:

- General hoisting support (which should immediately give us tensor constant folding when applied at the tile level)
- Adding SplitMain to the GPU pipelines, fix any bugs
- Breaking kernel launching in comp into compilation + launch so the compilation can be hoisted

Technically, this change could be separated into two PR's:  One to change stdx.pack/unpack semantics, and another to actually hook up init/fini, but I figure it's small enough to go in as one PR if no one disagrees.

**Review by flaub — DISMISSED** [review]

File: pmlc/dialect/tile/transforms/CMakeLists.txt [hunk 3]
```diff
@@ -9,10 +9,12 @@ pml_cc_library(
     pad_constraints.cc
     pad_ranges.cc
     padding.cc
+    split_main.cc
   DEPS
     LLVMSupport
     MLIRIR
     pmlc::dialect::tile::ir
+    pmlc::dialect::pxa::ir
```
Comment (flaub) [inline pmlc/dialect/tile/transforms/CMakeLists.txt:17]: Why do we depend on pxa? Seems like this should be stdx?

**Review by flaub — APPROVED** [review]

**Pull request merged**